### DISCUSSION TITLE
Can Next.js Commerce be easily modified to use my own API rather than BigCommerce's?

### DISCUSSION THREAD

**Discussion by RickRyan26** [vercel/next.js#18936] [Help] [answered]
Can Next.js Commerce be easily modified to use my own API rather than BigCommerce's? I haven't gotten a chance to dive into it yet myself but I assume it's not that difficult. Is there anything I should look out for? Thanks!

**Comment by jozsefsallai** [comment, marked as answer]
It does require some work, but it's nothing that can't be accomplished. They do plan on adding support for other e-commerce platforms in the future, but the current BigCommerce example works well too.

One thing to keep in mind is that the template uses a lot of stuff from the Storefront hooks made by BigCommerce themselves, so you won't find those hooks in the repo itself. You can still look at how they did it at https://github.com/bigcommerce/storefront-data-hooks and adapt some of the hooks to your own API.
